unit_tests/sources/declarative/partition_routers/test_cartesian_product_partition_router.py

```diff
@@ -5,8 +5,10 @@
 
 import pytest as pytest
 
+from airbyte_cdk.legacy.sources.declarative.incremental.datetime_based_cursor import (
```

Comment:
It feels like things should just be mocked and we could remove this but this seems like a non-trivial effort so can we just add a comment saying "screw this"?